general/style-guides/javascript.md

```diff
@@ -0,0 +1,190 @@
+---
+uid: style-guides-javascript
+title: JavaScript
+---
+
+# JavaScript
+
+## Tools
+
+Projects generally provide EditorConfig and ESLint configurations to help you adhere to this style guide.
+
+If one is not available or if you wish to start a new project, feel free to copy the existing files from one of the existing JavaScript projects.
+
+## File Structure
+
+### Filenames
+
+Filenames must be camel case and may not include underscores (`_`) or dashes (`-`). The filename's extensions must be `.js`.
+
+### General Structure
+
+All files must abide by the following general structure, with JSDoc comments being optional but preferred:
+
+```javascript
+/**
+ * This module documents the structure used by Javascript code in Jellyfin.
+ *
+ * @module path/to/this/module
+ */
+
+import module from 'dependency';
+import { myFunction, myClass } from 'dependency/submodule';
+import 'otherDependency';
+
+export const name = 'Some name';
+
+/**
+ * Defines a non-exported function, accessible only from this module.
+ *
+ * @param {Object} argument - The argument to pass to the function.
+ * @returns {Int|null} The resulting object from the function.
+ */
+function privateFunction (argument) {
+    // Code ommitted
+}
+
+export function publicFunction (argument) {
+    // Code ommitted
+}
+
+export default {
+    name,
+    publicFunction
+}
+
+```
+
+### Imports
+
+#### Naming
+
+If the imported module doesn't respect the file naming conventions, convert its name to camel case when importing its default export, like such:
```

Comment:
I'm really not a fan of ending sentences with colons.

Reply:
Currently, it is required at least in Web, since adding colons to everything recently actually fixed a few bugs...

Not using semi-colons is also error-prone since browsers WILL insert semicolons internally while evaluating (And so will transpilers and minifiers). These parsers have rules for how they parse statements which, if you don't know them, can cause issues.

For example, this:

```javascript
a = b + c 
(d + e).print() 
```

Would turn into this:

```javascript
a = b + c(d + e).print();
```

These issues can be hard to debug and, imo, in the interest of avoiding risk and ensuring we have less bugs introduced by accident, we should enforce semicolon usage.